### PR TITLE
fixes #503 - TOTP prompt no longer drops a newline

### LINKED ISSUE #503: TOTP prompt input echoing on a second line in CLI, rather than same one
:3

Screen Shot 2021-06-28 at 11 21 35

### PR DESCRIPTION
**Pull request by yaleman**
Fixes #503 

- [x] cargo fmt has been run
- [x] cargo clippy has been run
- [x] cargo test has been run and passes

### CODE REVIEW

File: kanidm_tools/src/cli/session.rs [hunk 1]
```diff
@@ -144,7 +144,8 @@ impl LoginOpt {
 
     fn do_totp(&self, client: &mut KanidmClient) -> Result<AuthResponse, ClientError> {
         let totp = loop {
-            println!("Enter TOTP: ");
+            print!("Enter TOTP: ");
+            io::stdout().flush().unwrap();
```
Comment (Firstyear) [inline kanidm_tools/src/cli/session.rs:148]: Curious what the stdout flush is for, but I trust it's needed :)

Comment (yaleman) [reply kanidm_tools/src/cli/session.rs:148]: Without the flush the app just hangs because stdout's like "yeah, I buffered it, now what?" 🙄

Comment (Firstyear) [reply kanidm_tools/src/cli/session.rs:148]: See I knew there would be a good reason :)

Comment (Firstyear) [reply kanidm_tools/src/cli/session.rs:148]: Actually, can you add that as a comment above that line so that future archelogists know this too?

Comment (yaleman) [reply kanidm_tools/src/cli/session.rs:149]: Done :)

**Pull request merged**